### DISCUSSION TITLE
Big trouble in relationship picker

### DISCUSSION THREAD

**Discussion by JCV67** [Budibase/budibase#12079] [Help] [answered]
Internal database | self hostted

When upgrading from v2.10.12 to any posterior version, until 2.11.33 relationship fields information is lost from forms (remains correct in db). It is strange that the information shows on some fields of the exact same type of relationship, but not in others. In one case I built a new column and wih one-to-many instead of many-to-many relationship, filled the rows one by one. In the table the information was correct (latest build), but in the form the information was "null".

Because of this, by now, I'm sucked in v2.10.12

![image](https://github.com/Budibase/budibase/assets/104722906/5345842a-db34-4f69-a90b-6ed6c2281051)

Thank you for your help

**Comment by melohagan** [comment]
Hey @JCV67 

Is that a default value, or is it a value coming from an update form?

Would you be able to share an app export with a minimal viable reproduction, and then I can take a look.

**Comment by JCV67** [comment]
Helos @melohagan, thank you for your reply. 

The value is from an update form. Do I post the export  .tar here?

**Reply by melohagan** [reply]
That works - just be sure to not include sensitive info, credentials, etc.

**Comment by JCV67** [comment]
Hi @melohagan, this is the discussion where I should had made my last comment:



https://github.com/Budibase/budibase/assets/104722906/7312580e-0945-4b8b-89ca-db8fc818dd78

**Comment by JCV67** [comment]
It might be related to he DP Repeater State filtering...


![RSpicker](https://github.com/Budibase/budibase/assets/104722906/b40455d2-9bb9-4db3-ae25-5f0243d3e201)

But I cannot find the problem

**Comment by JCV67** [comment]
If I add a copy of the same relationship picker to the form, the first one starts working correctly while the new one assumes the former behaviour of the first one.


![RSP](https://github.com/Budibase/budibase/assets/104722906/38dc2072-a26d-4323-86bf-3f16e2a21f83)

**Comment by JCV67** [comment]
Hi just found out that this does not happen to all clients, even others with several RS connections.
Is there a way to re-index database, or to perform other database maintenance operations over internal tables?

**Comment by JCV67** [comment]
Now I finally found out when this error occurs, it is similar to something that happened before. This happens  when the position of the record, by alphabetical order, is over 100 or so. When the chosen record is in the first 100, the form is correctly stored, after that we can see the error happening. It does no happen until the version 2.10.12 I cannot update my app.
thank you

**Reply by melohagan** [reply, marked as answer]
Hey @JCV67 There is an open issue for this bug as you have described here: https://github.com/Budibase/budibase/issues/11478

The relationship picker has gone through a few iterations and unfortunately this is a regression.

**Reply by JCV67** [reply]
Thank you @melohagan. I'll wait.

**Comment by JCV67** [comment]
Hello @melohagan, sorry for badgering. I don't know if this is a Budibase limitation, or if is something I am not doing as I should. Can you help me?
tnak you
